utils/bytes.go

```diff
@@ -7,8 +7,34 @@ import (
 	"crypto/rand"
 	"math/bits"
 	"sync"
+
+	"github.com/ava-labs/libevm/common"
 )
 
+// BytesToHashSlice packs [b] into a slice of hash values with zero padding
+// to the right if the length of b is not a multiple of 32.
+func BytesToHashSlice(b []byte) []common.Hash {
```

Comment:
1. Why do we do this multiple of `32` check? It seems like the api would be simpler if we did something like:

```
func BytesToHashSlice(b []byte) (common.Hash, error) {
```

2. We're adding a new exported package api here - would it make sense for add to add test coverage here?